src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -121,6 +121,9 @@ private void buildServiceMap() {
               singletonMap("ThreadSafe", "true"),
               "DEFAULT")
           .setSelfTest(LibCryptoRng.SPI.SELF_TEST);
+      // Following line is a workaround to ensure that the SecureRandom service
+      // is seens as ThreadSafe by SecureRandom, when using the alias name DEFAULT
```

Comment:
Is there a way to check that DEFAULT is still LibCryptoRng before setting ThreadSafe to true? Could there be a scenario where the DEFAULT becomes another provider that's not ThreadSafe and the following line is forgotten?
If there isn't a programmatic check, then maybe add a warning to the documentation, e.g.
"Warning: The following line assumes that DEFAULT is an alias of LibCryptoRng. Change it if the DEFAULT changes."

Reply:
Good point. I've added a check that DEFAULT points to an algorithm that is itself marked ThreadSafe.